src/Task/PhpCsFixerV2.php

```diff
@@ -72,7 +72,15 @@ public function run(ContextInterface $context)
         $arguments->addOptionalArgument('--allow-risky=%s', $config['allow_risky'] ? 'yes' : 'no');
         $arguments->addOptionalArgument('--cache-file=%s', $config['cache_file']);
         $arguments->addOptionalArgument('--config=%s', $config['config']);
-        $arguments->addOptionalCommaSeparatedArgument('--rules=%s', $config['rules']);
+
+        $rules = $config['rules'];
+        // JSON-encode rules if specified as a map.
+        if (array_values($rules) !== $rules) {
+            $arguments->addOptionalArgument('--rules=%s', json_encode($rules));
+        } else {
```

Comment:
Just to get some closure on the subject: this is how I see it: Adding the `implode()` method is not duplicating code.

The actual problem you want to solve is: When there are rules configured, I want to add them to the phpcsfixer cli tool.
A sub problem arises: I want to format the value in a different way based on the type: map or vector.

You want to solve the actual problem by solving the sub-problem in which you can use 2 different methods that are already existing.
Since there is no method available on the arguments collection that formats the string as a comma separated value for maps and json ecoded for vectors, you'll have to call 2 different methods on the arguments collection. This means that there are 2 possible paths in the task.

If you look at the problem from the "actual problem" point of view, you only want one path: When there are rules, I want to add an argument to the collection.
Since there is no method that can format based on map or vector, you'll have to write a new argument formatting implementation. This implementation is so custom, that there is no need for an extra method in the arguments collection. So you are not duplicating code, but you are writing a new implementation for the formatting problem. Sure, internally it uses a duplication of implode(), but that is only one part of the new formatting implementation.